buildDotnetModule: use platform-agnostic cp format

epetousis opened the pull request:
###### Description of changes

cp on macOS doesn't support the -T flag, which causes the fetch-deps script to fail. Appending `/.` to the source argument replicates the same functionality. Really minor change, but this was enough to break the script entirely on macOS.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

epetousis committed:
buildDotnetModule: use platform-agnostic cp format

cp on macOS doesn't support the -T flag, which causes the fetch-deps
script to fail. Appending `/.` to the source argument replicates the
same functionality.

IvarWithoutBones commented:
I wonder if it'd be useful to include coreutils manually, to avoid issues like these in the future. Using platform agonistic commands like this seem a bit fragile to me.

epetousis committed:
buildDotnetModule: use coreutils in fetch-deps

cp on macOS doesn't support the -T flag, which causes the fetch-deps
script to fail. Use Nix's coreutils to ensure the script works
consistently across all platforms.

epetousis commented:
Much better (and pretty obvious) idea. I've pushed a commit that switches all relevant commands used to their coreutils versions.

IvarWithoutBones commented on pkgs/build-support/dotnet/build-dotnet-module/default.nix:
```diff
@@ -118,23 +118,23 @@ in stdenvNoCC.mkDerivation (args // {
 
     fetch-deps = writeScript "fetch-${pname}-deps" ''
       set -euo pipefail
```
This was also an issue before this PR, but I think all the store paths end up making the generated script a bit difficult to read. I think we should add them all to `PATH` instead with something like this:
```suggestion
      set -euo pipefail

      export PATH="${lib.makeBinPath [ coreutils dotnet-sdk etc ]}"
```

That way we just generate one big line containing all the store paths, and the rest of the script should be a lot more readable. By unsetting `PATH` we also ensure we never accidentally use any dependencies not provided by the script itself.

IvarWithoutBones reviewed (changes requested):
This mostly looks good, I just have one more suggestion :)

IvarWithoutBones commented:
Could you rebase this?

epetousis commented:
@IvarWithoutBones have rebased and added the suggested changes. Apologies for the delay.

epetousis committed:
buildDotnetModule: set fetch-deps utils via PATH

IvarWithoutBones reviewed (approved):
I was successfully able to generate a lockfile with this on x86_64-darwin. Looks good to me, thanks!
